jcommune-view/jcommune-web-view/src/test/java/org/jtalks/jcommune/test/AdministrationControllerTest.groovy

```diff
@@ -51,70 +50,94 @@ class AdministrationControllerTest extends Specification {
+            groups.showGroupAdministrationPage(session = null)
+        then: 'The session is missing, perform login'
             thrown(MissingPropertyException)
     }
 
-    def 'must not be able to create group if you are not admin'() {
-        given: 'User created without access and login'
-            def user = new User()
-            users.created(user);
-            def session = users.signIn(user);
-        when: 'User creates group'
-            Group group = Groups.random()
-            groups.create(group,session)
-        then:
-            thrown(WrongResponseException)
+    def 'user without admin rights cannot create groups'() {
+        given: 'user logged in but doesn`t have admin rights'
+            def session = users.signInAsRegisteredUser(forum)
+        when: 'User attempts to create group'
+            def groupDto = Groups.randomDto()
+            groups.create(groupDto, session)
+        then: 'Access is denied'
+            thrown(AccessDeniedException)
     }
 
-    def 'test create group success'() {
-        given: 'User created and have admin permission on forum'
-            def user = new User()
-            users.created(user).withPermissionOn(forum, GeneralPermission.ADMIN)
-            def session = users.signIn(user)
-        when: 'User creates group'
-            Group group = Groups.random()
-            groups.create(group,session)
+    def 'only user with admin rights can create group'() {
+        given: 'User logged in and has admin rights'
+            def session = users.signInAsAdmin(forum)
+        when: 'User attempts to create group'
+            def groupDto = Groups.randomDto()
+            groups.create(groupDto,session)
         then: 'Group is created'
-            groups.isExist(group.name)
+            groups.isExist(groupDto.name)
+    }
+
+    @Ignore
+    def 'user without admin rights cannot edit groups'() {
+        given: 'user logged in but doesn`t have admin rights, random group created'
+            def session = users.signInAsRegisteredUser(forum)
+            def savedGroupId = groupsService.save(Groups.random())
+        when: 'User attempts to edit an existing group'
+            def groupDto = Groups.randomDto(id: savedGroupId)
+            groups.edit(groupDto, session)
+        then: 'Access is denied'
+            thrown(AccessDeniedException)
+    }
+
+    @Ignore
+    def 'only user with admin rights can edit editable group'() {
+        given: 'User logged in and has admin rights, random group created'
+            def session = users.signInAsAdmin(forum)
+            def savedGroupId = groupsService.save(Groups.random())
+        when: 'User attempts to edit an existing editable group'
+            def groupDto = Groups.randomDto(id: savedGroupId)
+            groups.edit(groupDto, session)
+        then: 'Group successfully edited'
+            groups.isExist(groupDto.name)
+        and: 'Edited group has the same ID as original'
```

Comment:
This sounds too technical. I'd get rid of this extra `and`and just put the code into previous `then`